### ISSUE TITLE
Files without marker and which are not read-only are not getting cleaned up

### ISSUE THREAD

**Issue by JohannesKonings** [projen/projen#3716]
In some scenarios it's necessary to format files after synthing or correct via eslint (e.g. ensure the order of JSON keys).
That's why the files are not created read-only.
That's not a problem if the files could be created with the projen marker.
But if it's not possible (e.g. renovate if not using json5) to set the projen marker, the files is not listedt in the files.json and will not be cleaned up if it's not sythed anymore.

In this example the file foo3.json is orphaned if this section will be removed: https://github.com/JohannesKonings/projen-files-test/blob/main/.projenrc.ts#L25

**Comment by mrgrain** [comment]
Yeah, that's somewhat by design. I'm not sure it's a good design though. 

Currently my recommendation is that projen generated files should be excluded from ESLint and "just generated correctly". But I'm aware that it's more effort and not always easily possible.

**Comment by JohannesKonings** [comment]
Could the files.json extended like the deps.json to store some metadata like the read-only status.
Maybe at project level or at the file components could be chosen if the not read-only files will be cleaned up.

**Comment by mrgrain** [comment]
> Could the files.json extended like the deps.json to store some metadata like the read-only status.

Read-only isn't the status, that's just the side effect of a file mansged but projen. But I agree that the current API for files turns this somewhat around. 

The problem that need solving, is that projen needs to detect if a file has changed. It does that by reading the current file contents and comparing it against the synth'd version. If there's a diff, the file has changed. 

In your scenario the file will always have changed. Which might be okay depending on the file.

**Comment by phoefflin** [comment]
Small helper to get rid of orphaned files: Remove everything with a projen marker and generate

```
rm -f $(git grep  -l 'Generated by projen' | grep  -E -v '(package.json|.projen)') && npx projen
```

a bit hacky and not perfect but catches at least most orphaned files for us

**Comment by JohannesKonings** [comment]
Another hacky approach.

The experimental ProjectTree shows metadata, which can be added e.g. via a Component. 
* Component which add the metadata: https://github.com/JohannesKonings/projen-files-test/blob/main/componentCleanupReadonlyFalse.ts#L10
* ProjectTree usage: https://github.com/JohannesKonings/projen-files-test/blob/main/.projenrc.ts#L56
* ProjectTree: https://github.com/JohannesKonings/projen-files-test/blob/main/.projen/tree.json

The Component also adds a Task which reads the metadata an delete all readonly = false files bevor the default task run: https://github.com/JohannesKonings/projen-files-test/blob/main/componentCleanupReadonlyFalse.ts#L19
This is the script for cleanup: https://github.com/JohannesKonings/projen-files-test/blob/main/cleanup-non-readonly-files.sh
